perllib/FixMyStreet/App/Controller/Auth.pm

```diff
@@ -44,10 +44,8 @@ sub general : Path : Args(0) {
 
     # decide which action to take
     $c->detach('code_sign_in') if $clicked_sign_in_by_code || ($data_email && !$data_password);
-    if (!$data_username && !$data_password && !$data_email) {
-        $c->detach('social/facebook_sign_in') if $c->get_param('facebook_sign_in');
-        $c->detach('social/twitter_sign_in') if $c->get_param('twitter_sign_in');
-        $c->detach('social/oidc_sign_in') if $c->get_param('oidc_sign_in');
+    if (!$data_username && !$data_password && !$data_email && $c->get_param('social_sign_in')) {
+        $c->detach('social/handle_sign_in');
```

Comment:
I think all the calls to `social/handle_sign_in` should be forward, not detach, that way the current behaviour of carrying on if bad data provided is maintained, I think.